From 05357ab798abb4dd5b8d0ed74f3046f430dfeac1 Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Mon, 21 Jan 2019 14:06:39 +0200
Subject: [PATCH 619/706] felix: fix null pointer on adding mdb entry

Adding a new entry in the bridge multicast DB generates a NPE
because the mcast db pointer was not initialized.
To reproduce the bug just issue the bellow cmd after creating
a bridge interface and adding swp0 to the bridge:

bridge mdb add dev switch port swp0 grp 224.0.2.1 vid 1

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
(cherry picked from commit 604978a1ca3ad6acd7de4658e9032a7c66f83120)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 769c5ca47212..68b1a2626f7e 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -1665,6 +1665,7 @@ int ocelot_init(struct ocelot *ocelot)
 	if (!ocelot->stats_queue)
 		return -ENOMEM;
 
+	INIT_LIST_HEAD(&ocelot->multicast);
 	ocelot_mact_init(ocelot);
 	ocelot_vlan_init(ocelot);
 
-- 
2.17.1

